lib/valueflow.cpp

```diff
@@ -4259,26 +4259,30 @@ struct LifetimeStore {
     }
 };
 
-static bool isOwningVariables(const std::list<Variable>& vars, int depth = 10)
+static bool hasBorrowingVariables(const std::list<Variable>& vars, const std::vector<const Token*>& args, int depth = 10)
 {
     if (depth < 0)
         return false;
-    return vars.empty() || std::all_of(vars.cbegin(), vars.cend(), [&](const Variable& var) {
-        if (var.isReference() || var.isPointer())
-            return false;
+    return std::any_of(vars.cbegin(), vars.cend(), [&](const Variable& var) {
         const ValueType* vt = var.valueType();
         if (vt) {
-            if (vt->pointer > 0)
+            if (std::none_of(args.begin(), args.end(), [vt](const Token* arg) {
+                return arg->valueType() && arg->valueType()->type == vt->type;
+            }))
                 return false;
-            if (vt->isPrimitive())
+            if (vt->pointer > 0)
                 return true;
-            if (vt->isEnum())
+            if (vt->reference != Reference::None)
                 return true;
+            if (vt->isPrimitive())
+                return false;
+            if (vt->isEnum())
+                return false;
             // TODO: Check container inner type
             if (vt->type == ValueType::CONTAINER && vt->container)
-                return !vt->container->view;
+                return vt->container->view;
             if (vt->typeScope)
-                return isOwningVariables(vt->typeScope->varlist, depth - 1);
+                return hasBorrowingVariables(vt->typeScope->varlist, args, depth - 1);
         }
         return false;
```

Comment:
This should return true. If the type is unknown is better to assume it will borrow as we will get better alias checking.